index.js

```diff
@@ -927,6 +931,17 @@ function nested (laterCode, name, key, schema, externalSchema, fullSchema, subKe
     case undefined:
       if ('anyOf' in schema) {
         schema.anyOf.forEach((s, index) => {
+          // inner levels of nesting
+          while (s.$ref) {
+            s = refFinder(s.$ref, fullSchema, externalSchema)
+          }
+          if ('anyOf' in s) {
+            s.anyOf.forEach((innerS) => {
+              if (innerS.$ref) {
+                throw new Error('Multiple nesting levels of anyOf not supported.')
```

Comment:
```suggestion
                throw new Error('Multiple nesting levels of anyOf and $ref are not supported.')
```

Reply:
I don't think this is an acceptable solution. The documentation states that the module uses JSON Schema -- https://github.com/fastify/fast-json-stringify/tree/a6652fd93a70656786f894f5c9b0d3a5d6eb4ce4#fastjsonstringifyschema . The JSON Schema spec does not limit the nesting level of references.

Reply:
I agree. But I prefer landing a 1-level solution that provides a meaningful error. Should we add "Please send a PR?"

Maybe we could add a comment before it?

Reply:
Okay, how about the error and an open issue asking for help?

Reply:
I'm sure you already discussed, but how about finding a way to pre-feed the whole schema to ajv - which then gets used only to process ifElse or anyOf - instead of extracting part of the schema and generating code with these ad-hoc logic?

ajv would have been happy if he had the whole schema from the start

Reply:
I don't think this is about ajv, but rather our internal code generation for the fast serialization.

Reply:
Ok today I read the fast-json-stringify internals for the first time so I may be mistaken, but what i detected is that the issue happens at `ajv.validate(schema, obj)` time.

Ajv is passed only a small portion of the schema, and fast-json-stringify takes care of fully 'de-referencing' all the $refs. From this the need to "follow the refs".

Reply:
Exactly! `fast-json-stringify` needs to have the full schema resolved before running. So, this should be fixable.

Reply:
See how this was solved for passing custom AJV instances into Fastify.

Reply:
I am just coming from that - custom ajv in fastify. Unfortunately I can't devote more time to the issue, I hope this PR gets accepted to at least provide some basic support.

If not, I'll hack my own npm package or some other work around.

Reply:
That’s a poor attitude to take. Matteo is trying to give feedback and guidance to get the PR accepted and merged.

Reply:
Been accepting feedback whole day yesterday - now I don't have any more time. 
I fail to grasp the poor attitude, but ¯\_(ツ)_/¯

Reply:
I happen to resurface a bit, should have some time to dedicate to this.

What is actually required for this PR to be accepted? I lost track

Reply:
We need to implement full recursive deferencing of the refs for fast-json-stringify to work properly.

Reply:
Actually turns out nested() func was already recursive enough :)
Multiple levels working